db/db.go

```diff
@@ -491,12 +539,29 @@ func defaultIndexFunc(d *DB) func(collection string, key ds.Key, oldData, newDat
 	}
 }
 
-func (d *DB) HandleNetRecord(rec net.ThreadRecord, key thread.Key, lid peer.ID, timeout time.Duration) error {
-	if rec.LogID() == lid {
-		return nil // Ignore our own events since DB already dispatches to DB reducers
+func (d *DB) ValidateNetRecordBody(_ context.Context, body format.Node, identity thread.PubKey) error {
```

Comment:
`Record`s that arrive over the `net` layer are presented to the connected app for validation against the collection's `WriteValidator`

Reply:
Nice. With this write validators, we should stress more than ever the single-writer approach.

That is, is very important that write validations relies on the assumption on getting writes in a single log where records are always presented in order. If some validation depends on two logs, depending on the order of processing you might have unwanted consequences.  Sharing thoughts to validate mentality.

Maybe under more advanced validators, multi-writers might make sense... but thinking about all possible out of order arrivals to validate or not sounds hard.

Reply:
Agree! Maybe I can add something to the readme that stresses this fact. I think a well written validator in many cases can avoid this problem when combined with the lower lever thread ACL... say, only the thread starter can alter some paths. Of course, this requires the user to think deeply about this stuff, which isn't ideal.

Reply:
Totally agree.